### DISCUSSION TITLE
AlphaPulldown v2.0.0

### DISCUSSION THREAD

**Discussion by DimaMolod** [KosinskiLab/AlphaPulldown#451] [Announcements]
### Release Notes

---

**🚀 Automated Workflows with Snakemake**  
Introduced a streamlined pipeline for modeling tasks, leveraging Snakemake for automation, scalability, and reproducibility. Installation is now simplified with Docker/Singularity containers.

**🔧 Unified Configuration Syntax**  
All AlphaPulldown modes now share a unified configuration syntax, simplifying input setup and enabling automatic sequence retrieval with UniProt IDs.

**🛠️ Modular Backend Support**  
Reorganized codebase to support multiple modeling backends flexibly. UniFold and AlphaLink2 are now integrated, with the potential to add more in the future.

**🔗 Cross-Link-Driven Modeling**  
AlphaLink2 integration supports modeling with cross-linking mass spectrometry (XL-MS) data, enhancing the accuracy of complex structural models.

**💾 Significant Storage Optimization**  
Achieved more than 90% reduction in storage through input feature and output file compression, promoting sustainability in large-scale modeling.

**📦 ModelCIF Format Support**  
Models can now be stored in ModelCIF format, aligning with FAIR principles for improved model accessibility and simpler model deposition in databases.

**🧩 Extended Modeling Capabilities**  
Increased flexibility with customizable modeling parameters, multimeric template support, and options to control MSA and template impact on predictions.

**📊 Enhanced Analysis Pipeline**  
Enriched the analysis toolkit with additional evaluation metrics like average pLDDT and PAE scores at protein interfaces for better assessment of model confidence..

**🔄 Improved Codebase and Documentation**  
Refactored code, introduced CI/CD pipelines, automated testing, and expanded documentation.

**🌐 Repository of Precomputed Features**  
Released a web-based [repository of precomputed input features](https://github.com/KosinskiLab/AlphaPulldown?tab=readme-ov-file#features-database) for multiple model organisms, reducing redundant computations and accelerating workflows.

---

**Comment by rukibuki** [comment]
Nice

Does "relax" work now? and what about all-vs-all does that work in this version, as in there are no redundancy in the tested protein pairs?

**Reply by DimaMolod** [reply]
Yes, "relax" works and all_vs_all [does not produce redundant complexes anymore](https://github.com/KosinskiLab/AlphaPulldown/blob/main/alphapulldown/utils/create_combinations.py#L27-L36). Please feel free to give it a try, and we are looking forward to your feedback, as usual

**Reply by rukibuki** [reply]
Sounds good! One random idea, what do you think about having a filter for the relax part? As in it only runs relax protocol on the best model (ranked 0) for a protein pair if the iptm is above a certain threshold, like 0,6 etc.?

**Reply by DimaMolod** [reply]
Yes, that is an interesting idea. We will think about implementing that, thanks!

**Reply by jkosinski** [reply]
Good idea, could you start a new issue with this feature request so we don’t forget?

**Comment by dongohhan** [comment]
As a wet lab researcher, what do you think about the option to annotate the residues involved in the interaction for high-scoring pairs, or so? 
When performing high-throughput analysis and reviewing each protein interaction, it would be necessary to identify which residues are involved in the interaction, and it would require a significant amount of manual labor to identify these interactions individually. I don't know if this would be feasible to implement, as I don't have any computational knowledge.

**Reply by jkosinski** [reply]
Hi, this is technically possible and we can think about it in the future. Could you start a new issue with this request so we remember? But note that you can get contact residues using many other packages including ChimeraX.
